Update endToEndEncryption.mdx (#1369)
streamr-dev/core-frontend: app/src/docs/content/streams/endToEndEncryption.mdx

import Warning from '$docs/components/Warning'

# End-to-end encryption

Confidentiality of events published on a stream can be guaranteed with end-to-end encryption. The publisher generates a AES-256 symmetric encryption key and encrypts the messages before publishing them to the network. As the publisher fully controls who can access their data, they are also responsible for communicating the key to subscribers - usually via the key exchange mechanism described below.

## Key exchange

> The subscribers need this symmetric group key in order to decrypt the data. They automatically obtain the key by performing a key exchange with the publisher. The key exchange happens using asymmetric encryption: 

Reviewer: ```suggestion
The subscribers need the symmetric group key in order to decrypt the data. They automatically obtain this key by performing a key exchange with the publisher. The key exchange happens using asymmetric encryption: 
```

- Both the publisher and subscriber generate a temporary RSA key pair to be used for the key exchange
- The subscriber sends a key request to the publisher, containing the subscriber's public key, signed with the subscriber's Ethereum key
- The publisher checks from the on-chain access control registry whether that subscriber should be able to access the stream, and if it does, the publisher responds with the AES symmetric key, encrypted with the publisher's RSA key for the subscriber's RSA key, and signs with the publisher's Ethereum key.

## Publisher liveness

> To perform key exchange with subscribers, the publisher must be online and present in the network. As the Streamr Network deals with real-time messages, publishers are often constantly online. Especially with old stored data it may happen, however, that the publisher has disappeared since publishing the data, making those messages inaccessible to subscribers who didn't receive the key at the time. This is a consequence of publishers being fully and solely in control of who can access their data on the network.

Reviewer: ```suggestion
To perform key exchange with subscribers, the publisher must be online and present in the Network. As the Streamr Network deals with real-time messages, publishers are often constantly online. However, it may happen that the publisher has disappeared since publishing the data, making those messages inaccessible to subscribers who have yet to receive the key. This is a consequence of the data publishers being fully in control of who can access their data on the Network.
```
